test/search/hex_client_test.exs

```diff
@@ -0,0 +1,107 @@
+defmodule Search.HexClientTest do
+  use ExUnit.Case, async: true
+
+  alias Search.HexClient
+
+  setup ctx do
+    tmp_dir = ctx[:tmp_dir]
+
+    if is_nil(tmp_dir) do
+      {:ok, []}
+    else
+      test_tar_contents = [{~c"README.md", "# I am a README!"}]
+      test_tar_path = Path.join(tmp_dir, "test_tar.tar.gz")
+
+      :ok =
+        :erl_tar.create(test_tar_path, test_tar_contents, [
+          :compressed
+        ])
+
+      {:ok, %{test_tar: test_tar_path, test_tar_contents: test_tar_contents}}
+    end
+  end
```

Comment:
I was recently doing similar stuff and remembered this PR, here's a quick way to create in-memory tar(.gz) without writing to disk:

```elixir
def create_tar(files, options \\ []) when is_list(files) do
  options = Keyword.validate!(options, compressed: false)
  compressed = Keyword.fetch!(options, :compressed)

  fun = fn
    :write, {pid, data} -> IO.write(pid, data)
    :position, {_pid, {:cur, 0}} -> {:ok, 0}
    :close, _pid -> :ok
  end

  {:ok, pid} = StringIO.open("")
  {:ok, tar} = :erl_tar.init(pid, :write, fun)

  for {path, content} <- files do
    :ok = :erl_tar.add(tar, content, to_charlist(path), [])
  end

  :ok = :erl_tar.close(tar)
  data = StringIO.flush(pid)
  if compressed, do: :zlib.gzip(data), else: data
end
```

it might be worth pursuing especially since we need the tarball in just one test and so this setup/1 block is otherwise unused. WDYT?